support/konnectivityproxy/proxy.go

```diff
@@ -114,7 +114,19 @@ func (p *konnectivityProxy) Dial(ctx context.Context, network string, requestAdd
 		return nil, fmt.Errorf("dialing proxy %q failed: %v", proxyAddress, err)
 	}
 
-	connectString := fmt.Sprintf("CONNECT %s HTTP/1.1\r\nHost: %s\r\n\r\n", requestAddress, "127.0.0.1")
+	requestHost, requestPort, err := net.SplitHostPort(requestAddress)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse request address %s: %w", requestAddress, err)
+	}
+	if p.resolveBeforeDial && !p.disableResolver && !isIP(requestHost) {
```

Comment:
It'd be nice to have a code comment similar to the one in the commit message close to the definition of this field

Reply:
Added the comment to the field description in the dialer options. Thx!